course3/homework-linkedlist/linkedlist.h

```diff
@@ -0,0 +1,129 @@
+#pragma once
+#include <iostream>
+#include <memory>
+
+namespace mau {
+
+// Reverse iterator. http://stackoverflow.com/questions/8542591/c11-reverse-range-based-for-loop
+template <typename T>
+struct reversion_wrapper { T& iterable; };
+
+template <typename T>
+auto begin (reversion_wrapper<T> w) { return rbegin(w.iterable); }
+
+template <typename T>
+auto end (reversion_wrapper<T> w) { return rend(w.iterable); }
```

Comment:
This will only work for Ts which are in the ```std``` namespace, as ```rbegin``` and ```rend``` are in the namespace ,and you don't specify it here. So the fact that it may work depends solely on "Argument Dependent Lookup": http://en.cppreference.com/w/cpp/language/adl